common/util/src/main/java/org/eclipse/dataspaceconnector/common/security/PemUtil.java

```diff
@@ -0,0 +1,48 @@
+/*
+ *  Copyright (c) 2022 Mercedes-Benz Tech Innovation GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Mercedes-Benz Tech Innovation GmbH - Initial API and Implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.common.security;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Base64;
+
+public final class PemUtil {
+    private static final String HEADER = "-----BEGIN CERTIFICATE-----";
+    private static final String FOOTER = "-----END CERTIFICATE-----";
+
+    public static X509Certificate readX509Certificate(@NotNull String encoded) throws GeneralSecurityException {
```

Comment:
We usually avoid to use exceptions, better to catch it and throw an `EdcException` (that's unchecked).